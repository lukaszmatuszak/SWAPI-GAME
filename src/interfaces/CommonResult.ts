export interface CommonResult {
    name: string;
    films: string[];
    created: string;
    edited: string;
    url: string;
}
