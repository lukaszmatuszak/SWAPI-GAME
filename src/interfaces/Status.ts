export enum Status {
    WINNER = 'winner',
    LOOSER = 'looser',
    DRAW = 'draw',
}
