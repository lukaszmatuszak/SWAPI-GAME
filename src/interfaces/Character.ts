import { CommonResult } from './CommonResult';

export interface Character extends CommonResult {
    height: string;
    mass: string;
    hair_color: string;
    skin_color: string;
    eye_color: string;
    birth_year: string;
    gender: string;
    homeworld: string;
    species: string[];
    vehicles: string[];
    starships: string[];
}
