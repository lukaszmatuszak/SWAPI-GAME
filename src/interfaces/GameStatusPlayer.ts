import { Status } from './Status';

export interface GameStatusPlayer {
    name: string;
    status: Status;
    mass: number;
    crew: number;
    score: number;
}
