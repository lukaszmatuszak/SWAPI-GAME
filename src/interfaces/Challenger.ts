import { Character } from './Character';
import { Starship } from './Starship';

export type Challenger = Character | Starship;
