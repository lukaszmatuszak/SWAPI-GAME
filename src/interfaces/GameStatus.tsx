import { GameStatusPlayer } from './GameStatusPlayer';

export interface GameStatus {
    playerOne: GameStatusPlayer;
    playerTwo: GameStatusPlayer;
}
