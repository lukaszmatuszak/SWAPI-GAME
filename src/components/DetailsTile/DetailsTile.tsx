import React from 'react';
import { Card, CardContent, Grid, Typography, styled } from '@material-ui/core';

import { GameStatusPlayer } from '../../interfaces/GameStatusPlayer';
import { Status } from '../../interfaces/Status';

const colors = {
    draw: '#b3b300',
    looser: '#990000',
    winner: '#006400',
};

const FullWidthGrid = styled(Grid)({
    width: '100%',
});

const StatusTypography = styled(Typography)(
    ({ status }: { status: Status }) => ({
        textTransform: 'uppercase',
        color: colors[status],
    }),
);

const RotatedImg = styled('img')({
    transform: 'rotate(45deg)',
});

const DetailsTile: React.FC<DetailsTileProps> = ({
    playerStatus,
}: DetailsTileProps) => (
    <Card>
        <CardContent>
            <Grid container alignItems="center" direction="column">
                <Grid item xs={12}>
                    <StatusTypography
                        variant="h5"
                        status={playerStatus.status}
                        data-test="status"
                    >
                        {playerStatus.status}
                    </StatusTypography>
                </Grid>
                <FullWidthGrid item xs={6}>
                    {playerStatus.crew ? (
                        <RotatedImg
                            width="100%"
                            src="falcon.svg"
                            alt="falcon"
                        />
                    ) : (
                        <img width="100%" src="vader.svg" alt="vader" />
                    )}
                </FullWidthGrid>
                <Grid item xs={6}>
                    <Typography data-test="details-name">
                        {playerStatus?.name}
                    </Typography>
                </Grid>
                <Grid item xs={6}>
                    <Typography
                        data-test={`details-${
                            playerStatus.crew ? 'crew' : 'mass'
                        }`}
                    >
                        {playerStatus.crew
                            ? `Crew: ${playerStatus.crew}`
                            : `Mass: ${playerStatus.mass}`}
                    </Typography>
                </Grid>
            </Grid>
        </CardContent>
    </Card>
);

interface DetailsTileProps {
    playerStatus: GameStatusPlayer;
}

export default DetailsTile;
