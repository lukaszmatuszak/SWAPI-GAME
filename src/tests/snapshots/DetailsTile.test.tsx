import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import DetailsTile from '../../components/DetailsTile/DetailsTile';
import { Status } from '../../interfaces/Status';

describe('DetailsTile Snapshot - Character ', () => {
    it('with winner status', () => {
        const wrapper = shallow(
            <DetailsTile
                playerStatus={{
                    name: 'test',
                    status: Status.WINNER,
                    mass: 20,
                    crew: null,
                    score: 1,
                }}
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Character with looser status', () => {
        const wrapper = shallow(
            <DetailsTile
                playerStatus={{
                    name: 'test',
                    status: Status.LOOSER,
                    mass: 20,
                    crew: null,
                    score: 1,
                }}
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Character with draw status', () => {
        const wrapper = shallow(
            <DetailsTile
                playerStatus={{
                    name: 'test',
                    status: Status.DRAW,
                    mass: 20,
                    crew: null,
                    score: 1,
                }}
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe('DetailsTile Snapshot - Starship', () => {
    it('with winner status', () => {
        const wrapper = shallow(
            <DetailsTile
                playerStatus={{
                    name: 'test',
                    status: Status.WINNER,
                    mass: null,
                    crew: 20,
                    score: 1,
                }}
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('with looser status', () => {
        const wrapper = shallow(
            <DetailsTile
                playerStatus={{
                    name: 'test',
                    status: Status.LOOSER,
                    mass: null,
                    crew: 20,
                    score: 1,
                }}
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('with draw status', () => {
        const wrapper = shallow(
            <DetailsTile
                playerStatus={{
                    name: 'test',
                    status: Status.DRAW,
                    mass: null,
                    crew: 20,
                    score: 1,
                }}
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
