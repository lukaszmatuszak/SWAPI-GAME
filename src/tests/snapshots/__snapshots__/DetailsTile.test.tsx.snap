// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsTile Snapshot - Character  Character with draw status 1`] = `
<WithStyles(ForwardRef(Card))>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      direction="column"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <Styled(WithStyles(ForwardRef(Typography)))
          data-test="status"
          status="draw"
          variant="h5"
        >
          draw
        </Styled(WithStyles(ForwardRef(Typography)))>
      </WithStyles(ForwardRef(Grid))>
      <Styled(WithStyles(ForwardRef(Grid)))
        item={true}
        xs={6}
      >
        <img
          alt="vader"
          src="vader.svg"
          width="100%"
        />
      </Styled(WithStyles(ForwardRef(Grid)))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <WithStyles(ForwardRef(Typography))
          data-test="details-name"
        >
          test
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <WithStyles(ForwardRef(Typography))
          data-test="details-mass"
        >
          Mass: 20
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;

exports[`DetailsTile Snapshot - Character  Character with looser status 1`] = `
<WithStyles(ForwardRef(Card))>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      direction="column"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <Styled(WithStyles(ForwardRef(Typography)))
          data-test="status"
          status="looser"
          variant="h5"
        >
          looser
        </Styled(WithStyles(ForwardRef(Typography)))>
      </WithStyles(ForwardRef(Grid))>
      <Styled(WithStyles(ForwardRef(Grid)))
        item={true}
        xs={6}
      >
        <img
          alt="vader"
          src="vader.svg"
          width="100%"
        />
      </Styled(WithStyles(ForwardRef(Grid)))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <WithStyles(ForwardRef(Typography))
          data-test="details-name"
        >
          test
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <WithStyles(ForwardRef(Typography))
          data-test="details-mass"
        >
          Mass: 20
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;

exports[`DetailsTile Snapshot - Character  with winner status 1`] = `
<WithStyles(ForwardRef(Card))>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      direction="column"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <Styled(WithStyles(ForwardRef(Typography)))
          data-test="status"
          status="winner"
          variant="h5"
        >
          winner
        </Styled(WithStyles(ForwardRef(Typography)))>
      </WithStyles(ForwardRef(Grid))>
      <Styled(WithStyles(ForwardRef(Grid)))
        item={true}
        xs={6}
      >
        <img
          alt="vader"
          src="vader.svg"
          width="100%"
        />
      </Styled(WithStyles(ForwardRef(Grid)))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <WithStyles(ForwardRef(Typography))
          data-test="details-name"
        >
          test
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <WithStyles(ForwardRef(Typography))
          data-test="details-mass"
        >
          Mass: 20
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;

exports[`DetailsTile Snapshot - Starship with draw status 1`] = `
<WithStyles(ForwardRef(Card))>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      direction="column"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <Styled(WithStyles(ForwardRef(Typography)))
          data-test="status"
          status="draw"
          variant="h5"
        >
          draw
        </Styled(WithStyles(ForwardRef(Typography)))>
      </WithStyles(ForwardRef(Grid))>
      <Styled(WithStyles(ForwardRef(Grid)))
        item={true}
        xs={6}
      >
        <Styled(img)
          alt="falcon"
          src="falcon.svg"
          width="100%"
        />
      </Styled(WithStyles(ForwardRef(Grid)))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <WithStyles(ForwardRef(Typography))
          data-test="details-name"
        >
          test
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <WithStyles(ForwardRef(Typography))
          data-test="details-crew"
        >
          Crew: 20
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;

exports[`DetailsTile Snapshot - Starship with looser status 1`] = `
<WithStyles(ForwardRef(Card))>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      direction="column"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <Styled(WithStyles(ForwardRef(Typography)))
          data-test="status"
          status="looser"
          variant="h5"
        >
          looser
        </Styled(WithStyles(ForwardRef(Typography)))>
      </WithStyles(ForwardRef(Grid))>
      <Styled(WithStyles(ForwardRef(Grid)))
        item={true}
        xs={6}
      >
        <Styled(img)
          alt="falcon"
          src="falcon.svg"
          width="100%"
        />
      </Styled(WithStyles(ForwardRef(Grid)))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <WithStyles(ForwardRef(Typography))
          data-test="details-name"
        >
          test
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <WithStyles(ForwardRef(Typography))
          data-test="details-crew"
        >
          Crew: 20
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;

exports[`DetailsTile Snapshot - Starship with winner status 1`] = `
<WithStyles(ForwardRef(Card))>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      direction="column"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <Styled(WithStyles(ForwardRef(Typography)))
          data-test="status"
          status="winner"
          variant="h5"
        >
          winner
        </Styled(WithStyles(ForwardRef(Typography)))>
      </WithStyles(ForwardRef(Grid))>
      <Styled(WithStyles(ForwardRef(Grid)))
        item={true}
        xs={6}
      >
        <Styled(img)
          alt="falcon"
          src="falcon.svg"
          width="100%"
        />
      </Styled(WithStyles(ForwardRef(Grid)))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <WithStyles(ForwardRef(Typography))
          data-test="details-name"
        >
          test
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <WithStyles(ForwardRef(Typography))
          data-test="details-crew"
        >
          Crew: 20
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;
